pandas/tests/frame/methods/test_transpose.py

```diff
@@ -103,3 +103,10 @@ def test_transpose_get_view_dt64tzget_view(self):
 
         rtrip = result._mgr.blocks[0].values
         assert np.shares_memory(arr._data, rtrip._data)
+
+    def test_transpose_mixed_dtypes(self):
```

Comment:
test the copy keyword. test an example which does keep object (e.g. (1, 1.5) and ( 'a', 1)) for (a, b)  and test with simple dtypes as well (e.g. int64)